contrib/automation_tests/core/orbit_e2e.py

```diff
@@ -0,0 +1,93 @@
+"""
+Copyright (c) 2020 The Orbit Authors. All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file.
+"""
+
+import logging
+from typing import Iterable, Type
+import time
+
+from absl import flags
+from pywinauto import Application
+
+import orbit_testing as ot
+from core.e2e_utils import first, OrbitE2EError
+
+flags.DEFINE_boolean('dev_mode', False, 'Dev mode - expect Orbit MainWindow to be opened, '
+                                        'and do not close Orbit at the end')
+
+
+class Fragment:
+    def __init__(self, test_case: "E2ETest"):
+        self._test_case = test_case
+
+    test_case = property(lambda self: self._test_case)
+    application = property(lambda self: self._test_case.application)
+
+    def window(self, force_update=False):
+        return self._test_case.window(force_update)
+
+    def execute(self):
+        pass
+
+    def expect_true(self, cond, description):
+        if not cond:
+            raise OrbitE2EError('Error executing test case %s, fragment %s. Condition did not hold: "%s"' %
+                                (self._test_case.name, self.__class__.__name__, description))
+
+    def expect_eq(self, left, right, description):
+        self.expect_true(left == right, description)
+
+
+class E2ETest:
+    def __init__(self, test_name: str, fragments: Iterable[Type[Fragment]], dev_mode: bool = False):
+        self._application = Application(backend='uia').connect(title_re='orbitprofiler')
+        self._test_name = test_name
+        self._dev_mode = dev_mode or flags.FLAGS.dev_mode
+        self._fragments = fragments[:]
+        self._window = None
+
+    name = property(lambda self: self._test_name)
+    application = property(lambda self: self._application)
+    dev_mode = property(lambda self: self._dev_mode)
+
+    def sleep_short(self):
```

Comment:
Since this base class provides general tooling for the tests I'd make the stuff from orbit_testing.py and also the new matcher member functions here.

Reply:
As discussed offline: Agreed, I'll do that in a separate PR so this one will not break the existing tests